### PR TITLE
 [HW-04] - Interactions between mentor and student

### PR DESCRIPTION
**Pull request by andreyruby**

### CODE REVIEW

File: HW04/andrey_bakanovsky/lib/mentor.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  def initialize(name, surname)
+    @name = name
+    @surname = surname
+    @notifications = []
+  end
+
+  def print_notification(notification)
+    puts "#{notification.read_status ? 'old' : 'new'} From #{notification.sender.surname} "\
+    "#{notification.sender.name[0]}: #{notification.text}"
+  end
```
Comment (B0RlS) [inline HW04/andrey_bakanovsky/lib/mentor.rb:16]: I think that a mentor shouldn't be doing this.

File: HW04/andrey_bakanovsky/lib/mentor.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    homework.mentor = self
+    homework.student = student
+    student.homeworks << homework
+    student.notifications << Notification.new("#{homework.status} #{homework.title} - #{homework.description} ",
+                                              self)
+    homework
+  end
+
+  def read_notifications!
+    puts "#{surname} #{name[0]}. mentor's notifications:"
```
Comment (B0RlS) [inline HW04/andrey_bakanovsky/lib/mentor.rb:29]:
```suggestion
```
Let's not enter the output of information into a method with logic

File: HW04/andrey_bakanovsky/lib/mentor.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    puts format(
+      ' %<a>6s | %<b>-30s | %<c>-20s | %<d>-20s | %<e>-30s ',
+      a: 'Title',
+      b: 'Description',
+      c: 'Mentor',
+      d: 'Student',
+      e: 'Status'
+    )
+    puts('-' * 116)
+  end
```
Comment (B0RlS) [inline HW04/andrey_bakanovsky/lib/mentor.rb:53]: it also seems to me that the mentor should not have this

File: HW04/andrey_bakanovsky/lib/mentor.rb (hunk 1 above)
Comment (andreyruby) [reply HW04/andrey_bakanovsky/lib/mentor.rb:16]: Yes, a mentor uses method read_notifications! which uses print_notification. I added method print_notification to private. Thanks.

File: HW04/andrey_bakanovsky/lib/mentor.rb (hunk 2 above)
Comment (andreyruby) [reply HW04/andrey_bakanovsky/lib/mentor.rb]: Ok, I put in special method as need for SOLID.

File: HW04/andrey_bakanovsky/lib/mentor.rb (hunk 3 above)
Comment (andreyruby) [reply HW04/andrey_bakanovsky/lib/mentor.rb:61]: The method I use in check_status_homeworks only. I added the method print_head_table to private.

**Pull request merged**